Switch otel to use environment variables to pickup trace context.

jdcasey opened the pull request:
The sidecar is intended to be run in short-lived builder pods which will inherit a trace context from the build orchestrator. This should be set on the builder pod using environment variables that are consistent with the Jenkins Opentelemetry Plugin, here:

https://github.com/jenkinsci/opentelemetry-plugin/blob/master/docs/job-traces.md#environment-variables-for-trace-context-propagation-and-integrations

They will be parsed using the same logic as the W3C trace context standard:

https://www.w3.org/TR/trace-context/#relationship-between-the-headers

To do this, we simply set the quarkus.opentelemetry.propagators configuration to include the 'envar' propagator, which is provided in the new opentelemetry-ext-cli-java dependency.

jdcasey commented:
The new dependency is hosted here: https://github.com/release-engineering/opentelemetry-ext-cli-java/

ligangty commented:
@jdcasey For that dep missing issue, we can set up the https://github.com/release-engineering/opentelemetry-ext-cli-java/ in our internal jenkins build to add the snapshot available in CI checking. But seems I don't have the permission to edit this repo, so you can do that if you want to.

jdcasey commented:
@ligangty I got it released at `org.commonjava.resilience.otel:opentelemetry-ext-cli-java:1.0.0` and it should show up on Maven Central any time now.

jdcasey commented:
@ligangty just after I updated this I got a response on my authorization request for com.redhat.resilience. So, I switched opentelemetry-ext-cli-java back over to the release-engineering org and re-released under the com.redhat.resilience.otel groupId. This should be its final location.
